lib/std/fs.zig

```diff
@@ -2232,6 +2232,45 @@ pub fn selfExePath(out_buffer: []u8) SelfExePathError![]u8 {
             // TODO could this slice from 0 to out_len instead?
             return mem.spanZ(@ptrCast([*:0]u8, out_buffer));
         },
+        .openbsd => {
+            // OpenBSD doesn't support getting the path of a running process, so try to guess it
+            if (os.argv.len >= 1) {
+                const argv0 = mem.span(os.argv[0]);
+                if (mem.indexOf(u8, argv0, "/") != null) {
+                    // argv[0] is a path (relative or absolute): use realpath(3) directly
+                    var real_path_buf: [MAX_PATH_BYTES]u8 = undefined;
+                    const real_path = try os.realpathZ(os.argv[0], &real_path_buf);
+                    if (real_path.len > out_buffer.len)
+                        return error.NameTooLong;
+                    mem.copy(u8, out_buffer, real_path);
+                    return out_buffer[0..real_path.len];
+
+                } else if (os.argv[0][0] != '\x00') {
+                    // argv[0] is not empty (and not a path): search it inside PATH
+                    const paths = std.os.getenv("PATH") orelse "";
+                    var path_it = mem.split(paths, ":");
+                    while (path_it.next()) |a_path| {
+                        var resolved_path_buf: [MAX_PATH_BYTES:0]u8 = undefined;
```

Comment:
FYI this is `MAX_PATH_BYTES+1` bytes wide, the sentinel counts as an extra byte.